GiniVision/Classes/Core/DocumentPickerCoordinator.swift

```diff
@@ -10,11 +10,39 @@ import Foundation
 import MobileCoreServices
 import Photos
 
-internal final class FilePickerManager: NSObject {
+protocol DocumentPickerCoordinatorDelegate: class {
+    /**
+     Called when a user picks one or several files from either the gallery or files explorer.
+     The completion might provide errors that must be handled here before dismissing the
+     pickers. It only applies for the `GalleryCoordinator` since on one side it is not possible
+     to handle the dismissal of UIDocumentPickerViewController and on the other side
+     the Drag&Drop is not done in a separate view.
+     
+     - parameter coordinator: `DocumentPickerCoordinator` where the documents were imported.
+     - parameter documents: One or several documents imported.
+     - parameter from: Picker used (either gallery, files explorer or drag&drop).
+     - parameter completion: `DocumentPickerCompletion` block used to check if there is an issue with
+     the captured documents. The completion block also has an inner block that is executed once the
+     picker has been dismissed when there are no errors.
+     */
+    func documentPicker(_ coordinator: DocumentPickerCoordinator,
+                        didPick documents: [GiniVisionDocument],
+                        from picker: DocumentPickerType,
+                        completion: DocumentPickerCompletion?)
+}
+
+public typealias DidDismissPickerCompletion = () -> Void
+public typealias DocumentPickerCompletion = (Error?, DidDismissPickerCompletion?) -> Void
```

Comment:
Completion within completion? It's getting out of hand!

Reply:
The naming can be improved.